hydroshare_jupyter_sync/models/resource_type_enum.py

```diff
@@ -0,0 +1,18 @@
+from enum import Enum
+
+
+class ResourceTypeEnum(Enum):
+    GenericResource = "GenericResource"
+    RasterResource = "RasterResource"
+    RefTimeSeriesResource = "RefTimeSeriesResource"
+    TimeSeriesResource = "TimeSeriesResource"
+    NetcdfResource = "NetcdfResource"
+    ModelProgramResource = "ModelProgramResource"
+    ModelInstanceResource = "ModelInstanceResource"
+    ToolResource = "ToolResource"
+    SWATModelInstanceResource = "SWATModelInstanceResource"
+    GeographicFeatureResource = "GeographicFeatureResource"
+    ScriptResource = "ScriptResource"
+    CollectionResource = "CollectionResource"
+    MODFLOWModelInstanceResource = "MODFLOWModelInstanceResource"
+    CompositeResource = "CompositeResource"
```

Comment:
Most of these resource types are deprecated and have been migrated to CompositeResource.  ToolResource is a web app connector and cannot hold files.  CollectionResource is the third type that is still used, all others have been migrated to Composite.  The script and model resources (ModelProgramResource, ModelInstanceResource, SWATModelInstanceResource, ScriptResource, MODFLOWModelInstanceResource) still exist in the system but cannot be created today.  Existing resources with that type will eventually be migrated to Composite.